pkg/pillar/containerd/containerd.go

```diff
@@ -69,77 +71,90 @@ const (
 var (
 	// default snapshotter used by containerd
 	defaultSnapshotter = "overlayfs"
-	ctrdCtx            context.Context
-	ctrdSystemCtx      context.Context
-	// CtrdClient is a handle to the current containerd client API
-	CtrdClient   *containerd.Client
-	contentStore content.Store
 )
 
-// InitContainerdClient initializes CtrdClient and ctrdCtx
-func InitContainerdClient() error {
-	log.Infof("InitContainerdClient")
-	var err error
-	if ctrdCtx == nil {
-		ctrdCtx = namespaces.WithNamespace(context.Background(), ctrdServicesNamespace)
-	}
-	if ctrdSystemCtx == nil {
-		ctrdSystemCtx = namespaces.WithNamespace(context.Background(), ctrdSystemServicesNamespace)
-	}
-	if CtrdClient == nil {
-		CtrdClient, err = containerd.New(ctrdSocket, containerd.WithDefaultRuntime(containerdRunTime))
-		if err != nil {
-			log.Errorf("InitContainerdClient: could not create containerd client. %v", err.Error())
-			return fmt.Errorf("initContainerdClient: could not create containerd client. %v", err.Error())
-		}
+var onceLock = &sync.Mutex{}
+var onceVal bool
+
+// once returns true the first time and then false
+func once() bool {
+	onceLock.Lock()
+	defer onceLock.Unlock()
+	if onceVal {
+		return false
+	} else {
+		onceVal = true
+		return true
 	}
-	if contentStore == nil {
-		contentStore = CtrdClient.ContentStore()
+}
+
+// Client is the handle we return to the caller
+type Client struct {
+	ctrdClient   *containerd.Client
+	contentStore content.Store
+}
+
+// NewContainerdClient returns a *Client
+// Callable from multiple go-routines.
+// createClient=false is used by tests where we just need the contexts etc
```

Comment:
The tail part of the comment (context) is no longer correct; the tests only need an empty Client struct.
That raises the question whether we should change the tests to not call this and instead manually create that empty Client struct, since that would remove the need fot this createClient argument.

Reply:
+1 to changing the tests

Reply:
Modified the test and the comment.